repository/src/main/java/org/pentaho/platform/repository2/unified/fs/FileSystemRepositoryFileDao.java

```diff
@@ -225,7 +228,10 @@ public RepositoryFile internalGetFile( File f ) {
       String jcrPath = f.getAbsolutePath().substring( rootDir.getAbsolutePath().length() );
       if ( jcrPath.length() == 0 ) {
         jcrPath = "/";
+      } else if ( IS_WINDOWS ) {
+        jcrPath = jcrPath.replaceAll( "\\\\", "/" );
       }
```

Comment:
We might want to consider using https://docs.oracle.com/javase/7/docs/api/java/io/File.html#separator instead of the string "/"

Reply:
Is the repository using the system path convention or always using "/" as the path separator? A cursory look at the code seems to point to the second case. Note that this is converting the _escaped_ windows path separator to the repository path representation.

Reply:
Makes sense. In that case, a couple of - better - approaches: 

- Use `RepositoryFile.SEPARATOR` instead of explicit `"/"` statements. 
  - And yes, our repository uses the unix-type separator as the convention
  - https://github.com/pentaho/pentaho-platform/blob/8.1.0.0-R/api/src/main/java/org/pentaho/platform/api/repository2/unified/RepositoryFile.java#L43
- Use `FilenameUtils.separatorsToUnix( jcrPath )`
  - https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FilenameUtils.html#separatorsToUnix(java.lang.String)
- Use `FilenameUtils.normalizeNoEndSeparator( jcrPath, true /* useUnixSeparator */ )`
  - https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FilenameUtils.html#normalizeNoEndSeparator(java.lang.String,%20boolean)


My $0.02: I'd steer away from any form of custom path separator handling.